### PR TITLE
refactor: A new dockerfile for the connector, restructuring of the codebase, sorting of the imports

### PR DESCRIPTION
**Pull request by Joel-hanson**
# Description

Reafactor the code to move all the command build related method to a different class

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
+package com.ibm.eventstreams.connect.jdbcsink.database.builder;
```
Comment (jhughes24816) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java:1]: need copy right headers

Comment (jhughes24816) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java:1]: Copy right headers are required at the top of new files and edited files will need updating

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+import org.apache.kafka.connect.data.Schema;
+
+import com.ibm.eventstreams.connect.jdbcsink.database.datasource.IDataSource;
+import com.ibm.eventstreams.connect.jdbcsink.database.writer.JDBCWriter;
+
+public class CommandBuilder {
+
+    public IDataSource dataSource = null;
+
+    public CommandBuilder(IDataSource connection) {
```
Comment (jhughes24816) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java:19]: These names are not quite lined up. Either the object is a data source or it is a connection to the data source? Would it be possible to clarify this?

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        }
+        return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+    }
+
+    public String buildInsertStatement(String tableName, List<String> fieldNames) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("INSERT INTO ").append(tableName);
+        sb.append("(").append(String.join(", ", fieldNames)).append(")");
+        sb.append(" VALUES ");
+        sb.append("(").append(String.join(", ", Collections.nCopies(fieldNames.size(), "?"))).append(")");
```
Comment (jhughes24816) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java:169]: Do we have any command builder tests? It is not clear why we are adding an n fold collection of "?" to this command some tests would make this class easier to understand

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java [hunk 4]
```diff
@@ -213,7 +102,7 @@ public void insert(String tableName, Collection<SinkRecord> records) throws SQLE
         try {
```
Comment (jhughes24816) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java:102]: `private String buildInsertStatement` should be removed?

**Review by jhughes24816 — CHANGES_REQUESTED** [review]
A couple of tidy up suggestions left

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java (hunk 3 above)
Comment (Joel-hanson) [reply src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java:169]: The preparestatement replaces those ? with values for example.

```java
PreparedStatement pstmt = con.prepareStatement("UPDATE EMPLOYEES
                                     SET SALARY = ? WHERE ID = ?");
   pstmt.setBigDecimal(1, 153833.00)
   pstmt.setInt(2, 110592)
```

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java (hunk 2 above)
Comment (Joel-hanson) [reply src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/builder/CommandBuilder.java]: I have split out class into command builder and database factory which makes it having individual behavior.

**Review by jhughes24816 — APPROVED** [review]
LGTM, nice work Joel!

**Pull request merged**